File: backend/src/main/java/com/roccia/backend/dto/ScoreResponse.java
package com.roccia.backend.dto;

import com.roccia.backend.domain.Score;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ScoreResponse {
    private Long id;
    private int sector;
    private int point;
    private LocalDateTime submittedAt;

    public static ScoreResponse from(Score score) {
        return ScoreResponse.builder()
                .id(score.getId())
                .sector(score.getSector())
                .point(score.getPoint())
                .submittedAt(score.getSubmittedAt())
                .build();
    }
}

File: frontend/lib/models/score_response.dart
class ScoreResponse {
  final int id;
  final int sector;
  final int point;
  final String submittedAt;

  ScoreResponse({
    required this.id,
    required this.sector,
    required this.point,
    required this.submittedAt,
  });

  factory ScoreResponse.fromJson(Map<String, dynamic> json) {
    return ScoreResponse(
      id: json['id'] as int,
      sector: json['sector'] as int,
      point: json['point'] as int,
      submittedAt: json['submittedAt'] as String,
    );
  }
}
